patient/models.py

```diff
@@ -0,0 +1,78 @@
+import hashlib
+
+from django.core.validators import MaxValueValidator, MinValueValidator
+from django.db import models
+
+from common.base_models import ImmutableBaseModel, CurrentBaseModel
+
+
+class Patient(CurrentBaseModel):
+    """
+    Patient record, ties all patient related info together.
+    """
+
+    identifier = models.CharField(max_length=50, null=False)
+    id_type = models.CharField(max_length=50, null=False)
+
+    class Meta:
+        ordering = ['-created']
+
+    @property
+    def patient_anon_id(self):
+        return hashlib.md5(self.id_type.encode() + self.identifier.encode()).hexdigest()[:12]
+
+    @property
+    def current_admission(self):
+        admission = self.admissions.first()
+        return admission
+
+
+class PersonalData(CurrentBaseModel):
```

Comment:
Can't we include these fields on the `Patient` model directly? E.g. we don't really need to have multiple date of births tied to a `Patient` as a foreign key IMO. This goes into the general sense of trying to have a "natural" API, i.e. it's smoother to write `patient.first_name` than `patient.personal_data.last().first_name`.

Reply:
We could. Basically I was trying to keep all editable data attributes at the same 'level'. That is as children of a `Patient`. Rather than having to remember that some data is in the patient itself, but some hangs off it.

In terms of a natural API, then you have all entries using `patient_id` as the key, rather than the patient (details) using the primary key id directly. Does that make sense?